mantidproject/mantid
# Fix bugs in PSI data loader

**Description of work.**
The PSI data loader did not work with the muon GUI. This PR fixes it along with a couple of small bugs.



**To test:**


Open the muon GUI
set the instrument to PSI
load some PSI dolly data (I can send the file)
It should all load correctly. 
The data should look like relatively flat for x<2. 
If you change the first good data (need to untick the box first) the plot will be cropped accordingly
Set the time zero to zero
Set the first good data to zero. 
The plot will show a flat bit followed by a large spike (at about x = first good data)
Check that the start date/time is before the end date/time


Load the multiperiod data
This has no group names and used to crash the GUI
It will now use default names of the form `group_specNum`


Fixes #27634 and #27381 . 




---
In release notes
#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
